src/App.tsx

```diff
@@ -1,51 +1,105 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
+import cn from 'classnames/bind';
 import './App.scss';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  { id: 1, name: 'Dumplings' },
+  { id: 2, name: 'Carrot' },
+  { id: 3, name: 'Eggs' },
+  { id: 4, name: 'Ice cream' },
+  { id: 5, name: 'Apple' },
+  { id: 6, name: 'Bread' },
+  { id: 7, name: 'Fish' },
+  { id: 8, name: 'Honey' },
+  { id: 9, name: 'Jam' },
+  { id: 10, name: 'Garlic' },
 ];
 
+interface Good {
+  id: number;
+  name: string;
+}
+
+const SORT_BY_ABC = 'abc';
+const SORT_BY_LENGTH = 'length';
+
+function getReorderedGoods(
+  goods: Good[],
+  sortType: string,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
```

Comment:
There seems to be a typo here. The spread operator should be used correctly to copy the array. It should be `const visibleGoods = [...goods];` instead of `const visibleGoods = [.goods];`.